--- a/core/fetcher/makeRequest.ts
+++ b/core/fetcher/makeRequest.ts
@@ -51,20 +51,20 @@
         signals.push(abortSignal);
     }
     const newSignals = anySignal(signals);
-    const response = await fetchFn(url, {
-        method: method,
-        headers,
-        body: requestBody,
-        signal: newSignals,
-        credentials: withCredentials ? "include" : undefined,
-        // @ts-ignore
-        duplex,
-        ...(disableCache && isCacheNoStoreSupported() ? { cache: "no-store" as RequestCache } : {}),
-    });
-
-    if (timeoutAbortId != null) {
-        clearTimeout(timeoutAbortId);
+    try {
+        return await fetchFn(url, {
+            method: method,
+            headers,
+            body: requestBody,
+            signal: newSignals,
+            credentials: withCredentials ? "include" : undefined,
+            // @ts-ignore
+            duplex,
+            ...(disableCache && isCacheNoStoreSupported() ? { cache: "no-store" as RequestCache } : {}),
+        });
+    } finally {
+        if (timeoutAbortId != null) {
+            clearTimeout(timeoutAbortId);
+        }
     }
-
-    return response;
 };
